share/functions/__fish_complete_mysql.fish

```diff
@@ -0,0 +1,25 @@
+function __fish_complete_mysql
+    set -l command $argv[1]
+
+    function __fish_mysql_query -a query
```

Comment:
I think these functions should be defined outside `__fish_complete_mysql`. They are put in the global scope either way.
Looks good otherwise.